app/src/main/java/com/github/se/polyfit/ui/components/list/MealList.kt

```diff
@@ -0,0 +1,98 @@
+package com.github.se.polyfit.ui.components.list
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.HorizontalDivider
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.github.se.polyfit.model.meal.Meal
+import com.github.se.polyfit.model.meal.MealOccasion
+import com.github.se.polyfit.model.nutritionalInformation.MeasurementUnit
+import com.github.se.polyfit.ui.theme.BorderPurple
+import com.github.se.polyfit.ui.theme.PrimaryPink
+import com.github.se.polyfit.ui.theme.PurpleGrey40
+import com.github.se.polyfit.viewmodel.meal.MealViewModel
+
+@Composable
+fun MealList(
+    meals: List<Meal>,
+    occasion: MealOccasion,
+    navigateTo: () -> Unit,
+    modifier: Modifier = Modifier,
+    mealViewModel: MealViewModel = hiltViewModel()
+) {
+  val filteredMeals = meals.filter { it.occasion == occasion }
+  val totalCalories = filteredMeals.sumOf { it.calculateTotalCalories() }.toString()
+
+  Surface(
+      modifier = modifier.fillMaxWidth().padding(8.dp, 0.dp).testTag("MealList"),
+      border = BorderStroke(1.dp, BorderPurple),
+      shape = RoundedCornerShape(4.dp),
+      color = Color.White) {
```

Comment:
I don't think our material theme matches the figma design. Maybe this can be a future ticket, but for now this and many other parts of the code base don't follow because it's not up to date with Figma's colors